File: Week_08-位运算-布隆过滤-排序/191.位-1-的个数.cpp
/*
 * @lc app=leetcode.cn id=191 lang=cpp
 *
 * [191] 位1的个数
 */

// @lc code=start
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int HmWeight = 0;
        for (int i=0; i<32; i++) {
            HmWeight += (n&1);
            n >>= 1;
        }
        return HmWeight;
    }
};
// @lc code=end

File: Week_08-位运算-布隆过滤-排序/191.位-1-的个数.py
#
# @lc app=leetcode.cn id=191 lang=python
#
# [191] 位1的个数
#

# @lc code=start
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        hwweight = 0
        for i in range(32):
            hwweight += (n&1)
            n >>= 1
        return hwweight
        
# @lc code=end

